.devops/deploy-pipelines.yml

```
@@ -0,0 +1,159 @@
+              command: 'buildAndPush'
+              tags: |
+                latest
+                $(Build.SourceVersion)
+                $(getAppVersion.appVersion)
+
+          - task: PublishPipelineArtifact@1
+            displayName: 'Publish manifests into pipeline artifacts'
+            condition: succeeded()
+            inputs:
+              targetPath: '$(Build.Repository.LocalPath)/helm'
+              artifact: 'helm'
+              publishLocation: 'pipeline'
+
+  - stage: stage_deploy
+    displayName: 'Deploy to ${{ variables.environment }} K8S'
+    dependsOn: ['stage_build']
+    variables:
+      appVersion: $[ stageDependencies.stage_build.job_build.outputs['getAppVersion.appVersion'] ]
+    condition: and(succeeded(), eq(variables.canDeploy, true))
+    jobs:
+      - deployment: job_deploy
+        displayName: 'Deploy'
+        pool:
+          name: $(selfHostedAgentPool)
+        environment: '$(environment)'
+        strategy:
+          runOnce:
+            deploy:
+              steps:
+                - download: none
+                - task: DownloadPipelineArtifact@2
+                  inputs:
+                    buildType: 'current'
+                    artifactName: 'helm'
+                    targetPath: '$(Pipeline.Workspace)/helm'
+                - task: KubectlInstaller@0
+                - task: Bash@3
+                  name: helm_dependency_build
+                  displayName: Helm dependency build
+                  inputs:
+                    workingDirectory: '$(Pipeline.Workspace)/helm'
+                    targetType: 'inline'
+                    script: |
+                      helm repo add pagopa-microservice https://pagopa.github.io/aks-microservice-chart-blueprint
+                      helm dep build
+                    failOnStderr: true
+                - bash: |
+                    echo 'microservice-chart:
+                       podAnnotations:
+                          "build/buildNumber": "$(Build.BuildNumber)"
+                          "build/appVersion": "$(appVersion)"
+                          "build/sourceVersion": "$(Build.SourceVersion)"' > buildMetadata.yaml
+                  displayName: Writing build metadata
+
+                - task: HelmDeploy@0
+                  displayName: Helm upgrade
+                  inputs:
+                    kubernetesServiceEndpoint: ${{ variables.kubernetesServiceConnection }}
+                    namespace: '$(deployNamespace)'
+                    command: upgrade
+                    chartType: filepath
+                    chartPath: $(Pipeline.Workspace)/helm
+                    chartName: ${{ variables.helmReleaseName }}
+                    releaseName: ${{ variables.helmReleaseName }}
+                    valueFile: "$(Pipeline.Workspace)/helm/values-${{ lower(variables.environment) }}.yaml"
+                    install: true
+                    waitForExecution: true
+                    arguments: --timeout 5m0s
```

add also buildMetadata values file generated inside the step "Writing build metadata"
```suggestion
                    arguments: --timeout 5m0s
                         --values buildMetadata.yaml
```